src/ellipsoidbump.c

```diff
@@ -6,239 +6,109 @@
-	float absmin = HUGE_VALF;
-
-	float guessingstep = 0.1f;
-	for (float x = -1; x <= 1; x += guessingstep) {
-		float f = (A*x + B)*sqrtf(C*x*x + D) + E*x;
-		if(!isfinite(f)) {
-			log_printf("got f(x) = %.10f. Arguments: A=%.10f B=%.10f C=%.10f D=%.10f E=%.10f",
-				f, A, B, C, D, E);
-			continue;
+	SDL_assert(a <= b);
+
+	// https://en.wikipedia.org/wiki/Golden-section_search
+	// a < t < u < b (if a=b, loop below does nothing)
+	float phi = 1.618033988749895f;
+	float t = a + (2-phi)*(b-a);
+	float u = a + (-1+phi)*(b-a);
+
+#define f(x) (sqrtf(A1*x*x + B1*x + C1) + sqrtf(A2*x*x + B2*x + C2))
+	float ft = f(t);
+	float fu = f(u);
+
+	int iter = 0;  // just in case of nan weirdness, lol
+	while (b-a > 1e-3f) {
+		if (iter++ == 25) {
+			log_printf(
+				"hitting max number of iterations: "
+				"f(x) = sqrt(%fx^2 + %fx + %f) + sqrt(%fx^2 + %fx + %f) on [%f,%f]",
+				A1,B1,C1,A2,B2,C2,a,b);
+			break;
 		}
-
-		float absval = fabsf(f);
-		if (absval < absmin) {
-			absmin = absval;
-			xguess = x;
+		if (ft < fu) {
+			// Increases between t and u, max can't be on [a,t]
+			a = t;
+			t = u;  // magic constants using phi chosen so that this works
+			ft = fu;
+			u = a + (-1+phi)*(b-a);
+			fu = f(u);
+		} else {
+			b = u;
+			u = t;
+			fu = ft;
+			t = a + (2-phi)*(b-a);
+			ft = f(t);
 		}
 	}
-
-	/*
-	Now newton's method to refine our guess quickly. Works well most of the time,
-	but some special cases occur if you jump a lot with both players by smashing
-	buttons on keyboard.
-	*/
-	float x = xguess;
-	int iter = 0;
-	float sub;
-	do {
-
-#define LOG(MSG) \
-	log_printf("%s (iter=%d x=%.10f xguess=%.10f A=%.10f B=%.10f C=%.10f D=%.10f E=%.10f)", \
-	(MSG), iter, x, xguess, A, B, C, D, E)
-
-		/*
-		Could use a squared-both-sides version of the equation here to avoid
-		calculating square roots, but this works quickly enough
-		*/
-		float sqrtstuff = sqrtf(C*x*x + D);
-		float sqrtstuff_derivative = C*x/sqrtstuff;
-		float f = (A*x + B)*sqrtstuff + E*x;
-		float f_derivative = A*sqrtstuff + (A*x + B)*sqrtstuff_derivative + E;
-		if (f_derivative == 0) {
-			LOG("derivative is zero");
-			return xguess;
-		}
-
-		// newton's method formula from high school: x_(n+1) = x_n - f(x_n)/f'(x_n)
-		// this doesn't behave nicely if derivative is zero or small
-		sub = f / f_derivative;
-		if (!isfinite(sub)) {
-			LOG("division by nonzero derivative gave something weird");
-			return xguess;
-		}
-		x -= sub;
-
-		if (fabs(x - xguess) > 2*guessingstep) {
-			LOG("x value is far away from guess");
-			return xguess;
-		}
-		if (fabsf(x) > 1) {
-			if (fabsf(x) > 1.01)
-				LOG("x value not between -1 and 1");
-			return xguess;
-		}
-
-		// it typically takes 3 or 4 iterations to make it precise, so this limit is plenty
-		if (++iter == 20) {
-			LOG("hitting max number of iterations");
-			break;
-		}
-#undef LOG
-	} while (fabsf(sub) > PRECISION_REQUIREMENT);
-
-	return x;
+	return (ft+fu)/2;
```

Comment:
max value typically bumps a bit bigger in the middle

```suggestion
	return f((t+u)/2);
```